### PR TITLE
Fixes so that more JUCE plugs pass clap-validator

### PR DESCRIPTION
**Pull request by baconpaul**
1. Implment write in chunks as opposed to assume full write works
2. Implmenet paramFlush properly; refactor events slightly to avoid
   duplicatin in doing so.

### CODE REVIEW

**Comment by baconpaul** [comment]
This will also close #45

**Comment by baconpaul** [comment]
@jatinchowdhury18 these changes do the two things mentioned, but we still don't pass the validator with monique or surge. (Surge has upstream changes too for the validator). I'm talking to robbert about that, but I think these changes are good

One f the problems the validator has is it doesn't seem to respect the cookie properly so I added more asserts to just be 100 sure we set up the cookie properly.

File: src/wrapper/clap-juce-wrapper.cpp [hunk 1]
```diff
@@ -1298,8 +1319,19 @@ class ClapJuceWrapper : public clap::helpers::Plugin<
 
         processor->getStateInformation(chunkMemory);
 
-        auto written = stream->write(stream, chunkMemory.getData(), chunkMemory.getSize());
-        return written == (int64_t)chunkMemory.getSize();
+        auto dat = (uint8_t *)chunkMemory.getData();
+        auto sz = chunkMemory.getSize();
+        while (sz > 0)
```
Comment (jatinchowdhury18) [inline src/wrapper/clap-juce-wrapper.cpp]:
```suggestion
        while (sz >= 0)
```
I think?

**Review by jatinchowdhury18 — APPROVED** [review]
Looks good! Great that we have `paramsFlush()` implemented now as well.

**Comment by baconpaul** [comment]
OK @jatinchowdhury18 i think this one is ready for review when you get a chance!

**Pull request merged**